src/CitizenAction/CitizenActionMessageNotifier.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace AppBundle\CitizenAction;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\CitizenAction;
+use AppBundle\Entity\EventRegistration;
+use AppBundle\Events;
+use AppBundle\Mailer\MailerService;
+use AppBundle\Mailer\Message\CitizenActionCancellationMessage;
+use AppBundle\Repository\EventRegistrationRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class CitizenActionMessageNotifier implements EventSubscriberInterface
+{
+    private $mailer;
+    private $urlGenerator;
+    private $registrationRepository;
+
+    public function __construct(
+        MailerService $mailer,
+        UrlGeneratorInterface $urlGenerator,
+        EventRegistrationRepository $registrationRepository
+    ) {
+        $this->mailer = $mailer;
+        $this->urlGenerator = $urlGenerator;
+        $this->registrationRepository = $registrationRepository;
+    }
+
+    public function sendCancellationEmail(CitizenActionEvent $citizenActionEvent): void
+    {
+        if (!$citizenActionEvent->getCitizenAction()->isCancelled()) {
+            return;
+        }
+
+        $subscriptions = $this->registrationRepository->findByEvent($citizenActionEvent->getCitizenAction());
+
+        if (count($subscriptions) > 0) {
+            $registrationChunks = array_chunk($subscriptions->toArray(), MailerService::PAYLOAD_MAXSIZE);
+
+            foreach ($registrationChunks as $chunk) {
+                $this->mailer->sendMessage($this->createCancelMessage(
+                    $chunk,
+                    $citizenActionEvent->getCitizenAction(),
+                    $citizenActionEvent->getAuthor()
+                ));
+            }
+        }
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            Events::CITIZEN_ACTION_CANCELLED => ['sendCancellationEmail'],
+        ];
+    }
+
+    private function createCancelMessage(array $registered, CitizenAction $citizenAction, Adherent $host): CitizenActionCancellationMessage
```

Comment:
`$host` (Author) is nullable and optional in `CitizenActionEvent` constructor

Reply:
Done with `CitizenActionCancelledEvent`.